e2e/playwright/tests/todo_plugin.spec.ts

```diff
@@ -8,35 +8,71 @@
 
 import {expect, test} from '@e2e-support/test_fixture';
 import SlashCommandSuggestions from 'support/components/slash_commands';
-import {fillMessage, getTodoBotDMPageURL} from 'support/utils';
+import {fillMessage, getLastPost, getTodoBotDMPageURL, postMessage, } from 'support/utils';
+
+test.beforeEach(async ({ page, pw }) => {
+  const {adminClient, adminUser} = await pw.getAdminClient();
+  if (adminUser === null) {
+    throw new Error('can not get adminUser');
+  }
+  const dmURL = await getTodoBotDMPageURL(adminClient, '', adminUser.id);
```

Comment:
Can we change this to be:

```suggestion
  const dmURL = await getBotDMPageURL(adminClient, adminUser.id, 'todo');
```

Having any todo plugin specific stuff in the support files will make it more difficult to refactor to shared code. If something is specific to the todo plugin, I think it should be in a file that contains only todo plugin specific things

Note that I removed the `teamName` argument, and added a `botUsername` argument. I think the `teamName` argument was a bit cryptic, since we are usually just passing an empty string it seems. 0/5 on how much value that provides. We do an extra call to the API to get a valid team name when that's not provided, so that's not ideal. Maybe that's something that be calculated ahead of time before any tests are run somehow

Reply:
Makes sense. I've reorganized the code, could you review it once? @mickmister